src/common/wallet/store.ts

```diff
@@ -68,10 +68,33 @@ export function decryptMnemonic(encryptedMnemonicWithSalt: string, password: str
   }).toString(CryptoJS.enc.Utf8);
 }
 
-export const getWallet = (): Wallet | null => {
+export const getCloudStorageItem = (store: string): Promise<string> => {
+  const CloudStorage = window.Telegram?.WebApp?.CloudStorage;
+
+  return new Promise((resolve, reject) => {
+    if (!CloudStorage) reject('CloudStorage is not available');
+
+    window.Telegram.WebApp.CloudStorage.getItem(store, (err, value) => {
+      if (err || !value) {
+        reject(err);
+      }
+
+      resolve(value!);
+    });
+  });
+};
+
+export const getWallet = async (): Promise<Wallet | null> => {
   const publicKey = localStorage.getItem(getStoreName(PUBLIC_KEY_STORE));
+  const backupLocalDate = localStorage.getItem(getStoreName(BACKUP_DATE));
+  try {
+    const backupCloudDate = await getCloudStorageItem(BACKUP_DATE);
```

Comment:
I think it shouldn't be fetching as part of getting the local wallet because if cloud is not available then we always route a user to the onboarding